src/test/java/io/gingersnapproject/testcontainers/annotation/WithDatabase.java

```diff
@@ -0,0 +1,26 @@
+package io.gingersnapproject.testcontainers.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import io.gingersnapproject.testcontainers.BaseGingersnapResourceLifecycleManager;
+import io.gingersnapproject.testcontainers.DatabaseProvider;
+import io.gingersnapproject.testcontainers.Profiles;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(Profiles.ProfileCondition.class)
+@QuarkusTestResource(value = BaseGingersnapResourceLifecycleManager.class, restrictToAnnotatedClass = true, parallel = true)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface WithDatabase {
+
+   KeyValue[] properties() default {};
+
+   Class<? extends DatabaseProvider> value() default DatabaseProvider.class;
+
+   String[] rules() default {};
```

Comment:
Other than verifying that the rule format is correct, it seems like these values are never used.

Reply:
We use the `rules()` to be able to define rules per test:

https://github.com/jabolina/db-syncer/blob/adf4d18d720583ffe4034004f63e8684122b8a0a/src/test/java/io/gingersnapproject/testcontainers/BaseGingersnapResourceLifecycleManager.java#L81-L87

We do this instead of using the properties file defining all the rules.